databricks/sdk/core.py

```diff
@@ -1067,6 +1056,82 @@ def do(self,
         if headers is None:
             headers = {}
         headers['User-Agent'] = self._user_agent_base
+        retryable = retried(timeout=timedelta(seconds=self._retry_timeout_seconds),
+                            is_retryable=self._is_retryable)
+        return retryable(self._perform)(method,
+                                        path,
+                                        query=query,
+                                        headers=headers,
+                                        body=body,
+                                        raw=raw,
+                                        files=files,
+                                        data=data)
+
+    @staticmethod
+    def _is_retryable(err: BaseException) -> Optional[str]:
+        # this method is Databricks-specific port of urllib3 retries
+        # (see https://github.com/urllib3/urllib3/blob/main/src/urllib3/util/retry.py)
+        # and Databricks SDK for Go retries
+        # (see https://github.com/databricks/databricks-sdk-go/blob/main/apierr/errors.go)
+        from urllib3.exceptions import ProxyError
+        if isinstance(err, ProxyError):
+            err = err.original_error
```

Comment:
Do we need to do `return _is_retryable(err.original_err)` in case of deeper nesting of proxy errors?

Reply:
ProxyError wraps the ConnectionError and other things

Reply:
```python
new_err = ProxyError(
        f"Unable to connect to proxy"
        f"{http_proxy_warning if is_likely_http_proxy and proxy_scheme == 'https' else ''}",
        err,
    )
```

it's needed for cases when HTTP/HTTPS proxy is not temporary unavailable and retry on those as well

Reply:
Makes sense, I just wonder if it will ever be more deeply nested.

Reply:
not from the current code paths